hydrator-common/src/main/java/io/cdap/plugin/common/Asset.java

```diff
@@ -15,25 +15,33 @@
  */
 package io.cdap.plugin.common;
 
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
 /**
  * Represents a dataset with a FQN which is a fully-qualified unique identifier for a dataset
  * and the location of the dataset.
  */
 public class Asset {
 
-  public static final String DEFAULT_LOCATION = "unknown";
+  private static final String DEFAULT_LOCATION = "unknown";
 
+  private final String referenceName;
   private final String fqn;
   private final String location;
 
-  public Asset(String fqn) {
-      this.fqn = fqn;
-      this.location = DEFAULT_LOCATION;
+  private Asset(String referenceName, @Nullable String fqn, @Nullable String location) {
```

Comment:
Can `referenceName` be null here? I think we will make it optional for some plugins right?

Reply:
So this will be either the reference name or the normalized FQN.
The plugins will check if reference name is present or not - if yes it will send the same, if not it will send the normalized fqn.